src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassReader.java

```diff
@@ -621,7 +633,24 @@ public Type getEnclosingType() {
                         public void setEnclosingType(Type outer) {
                             throw new UnsupportedOperationException();
                         }
-                    };
+
+                        @Override
+                        public List<Type> getTypeArguments() {
+                            List<Type> formalsCp = ((ClassType)t.type.tsym.type).typarams_field;
```

Comment:
This works. You might use another boolean flag up top to say whether you actually need to adjust the type arguments (which is only needed the first time) ?

Reply:
sure sorry I forgot about that